lxd/images.go

```diff
@@ -167,29 +167,29 @@ func imgPostContInfo(d *Daemon, r *http.Request, req imageFromContainerPostReq,
 		return info, err
 	}
 
-	if err := c.exportToDir(snap, builddir); err != nil {
-		return info, err
-	}
-
 	// Build the actual image file
-	tarfname := fmt.Sprintf("%s.tar.xz", name)
+	tarfname := fmt.Sprintf("%s.tar", name)
 	tarpath := filepath.Join(builddir, tarfname)
-	args := []string{"-C", builddir, "--numeric-owner", "-Jcf", tarpath}
-	if shared.PathExists(filepath.Join(builddir, "metadata.yaml")) {
-		args = append(args, "metadata.yaml")
+	tarfile, err := os.OpenFile(tarpath, os.O_WRONLY|os.O_CREATE|os.O_TRUNC, 0600)
+	if err != nil {
+		return info, err
 	}
-	args = append(args, "rootfs")
-	output, err := exec.Command("tar", args...).CombinedOutput()
+	if err := c.exportToTar(snap, tarfile); err != nil {
+		tarfile.Close()
+		return info, fmt.Errorf("imgPostContInfo: exportToTar failed: %s\n", err)
+	}
+	tarfile.Close()
+
+	args := []string{tarpath}
+	_, err = exec.Command("gzip", args...).CombinedOutput()
```

Comment:
Oh, good point.  I guess bc xz wasn't supported by golang's native compress/* libraries.  May as well go back to using xz indeed.